### PR TITLE
Add basic profiling support

### PR DESCRIPTION
**Pull request by atgreen**
This patch adds basic profiling support.  The simulator will record call arc and basic block counts in gprof compatible files.

The basic block counts are imperfect because they only count branch/jmp/jsra/jsra targets as basic blocks at runtime, and will omit the first entry into a basic block through straightline code. gprof basic block counts are usually handled through code instrumentation, which I didn't want to have to require.

The new code increased moxie.cc to over 1k LOC so I simplified the branch code to shrink it back. :)

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -3,6 +3,8 @@ noinst_PROGRAMS = sandbox gen-sum
 
 noinst_LIBRARIES = libsandbox.a
 
+AM_CXXFLAGS = -std=c++11
```
Comment (jgarzik) [inline src/Makefile.am]: Is c++11 required?  We have avoided this in the main bitcoin codebase, and my default preference is to be as liberal WRT old compilers as possible.

**Comment by atgreen** [comment]
Yes, it's required for std::unordered_map.  Maybe there's some other libstdc++ type I can use...

**Comment by jgarzik** [comment]
In general looks good - concept ACK - thanks @atgreen

Please rebase and force-push when you have a moment.  That will fix merge conflicts, and trigger the newly-enabled Travis CI pull testing to ensure your change builds & tests.

**Comment by jgarzik** [comment]
Oh and also:  Please put branch simplification code into a separate commit (same pull request is fine).  Trying hard to follow standard git protocol and separating out logical commits.

Thanks!
